File: se/se6029/mob-programming/self-practice/01-guinea-pig-car-car-cpp/src/car.cpp
// car.cpp

#include "car.h"

Car::Car() {
    speed = 0;
    initSpeed = speed;
    accelerateSpeed = 0;
}

Car::Car(int initSpeed, int accelerateSpeed) {
    speed = initSpeed;
    this->initSpeed = initSpeed;
    this->accelerateSpeed = accelerateSpeed;
}

void Car::accelerate() {
    speed = accelerateSpeed;
}

void Car::resetSpeed() {
    speed = initSpeed;
}

int Car::getSpeed() {
    return speed;
}

File: se/se6029/mob-programming/01-guinea-pig-car-car/src/Car/Car.java
// Car/Car.java

package Car;

public class Car {
    protected int speed;
    protected int initSpeed;
    protected int accelerateSpeed;

    // 為避免沒有給任何值
    public Car() {
        this.speed = 0;
        this.initSpeed = speed;
        this.accelerateSpeed = 0;
    }

    public Car(int initSpeed, int accelerateSpeed) {
        this.speed = initSpeed;
        this.initSpeed = initSpeed;
        this.accelerateSpeed = accelerateSpeed;
    }

    public void accelerate() {
        this.speed = this.accelerateSpeed;
    }

    public void resetSpeed() {
        this.speed = this.initSpeed;
    }

    public int getSpeed() {
        return speed;
    }
}
